kafka-producer/src/main/java/com/expedia/www/haystack/pipes/kafka/producer/Service.java

```diff
@@ -0,0 +1,131 @@
+import java.util.Map;
+import java.util.stream.Collectors;
+
+
+public class Service {
+
+    private static final SerdeFactory serdeFactory = new SerdeFactory();
+    private static final MetricRegistry metricRegistry = new MetricRegistry();
+    private static final ProjectConfiguration projectConfiguration = ProjectConfiguration.getInstance();
+    @VisibleForTesting
+    static Logger logger = LoggerFactory.getLogger(Service.class);
+    @VisibleForTesting
+    static HealthController healthController = new HealthController();
+    @VisibleForTesting
+    static Service service = null;
+    @VisibleForTesting
+    private static List<KafkaProducerExtractorMapping> extractorListMap = null;
+
+    private Service() {
+    }
+
+    public static void main(String[] args) {
+        logger.info("Initializing Kafka Consumers");
+        service = Service.getInstance();
+        service.inPlaceHealthCheck();
+        JmxReporter jmxReporter = service.getJmxReporter();
+        jmxReporter.start();
+        final ProtobufToKafkaProducer protobufToKafkaProducer = service.getProtobufToKafkaProducer(service.getKafkaStreamStarter());
+        protobufToKafkaProducer.main();
+    }
+
+    public static Service getInstance() {
+        if (null == service) {
+            service = new Service();
+        }
+        return service;
+    }
+
+    public static List<KafkaProducerExtractorMapping> getKafkaProducerExtractorMapping(ProjectConfiguration projectConfiguration) {
+
+        if (null == extractorListMap) {
+            List<KafkaProducerConfig> kafkaProducerConfigs = projectConfiguration.getKafkaProducerConfigs();
+            KafkaToKafkaPipeline.Factory factory = new KafkaToKafkaPipeline.Factory();
+
+            Map<String, KafkaProducerWrapper> kafkaProducerNameMap = kafkaProducerConfigs.stream()
+                    .collect(Collectors.toMap(kafkaProducerConfig -> kafkaProducerConfig.getName(),
+                            kafkaProducerConfig -> {
+                                KafkaProducer<String, String> kafkaProducer = factory.createKafkaProducer(kafkaProducerConfig.getConfigurationMap());
+                                KafkaProducerMetrics kafkaProducerMetrics = new KafkaProducerMetrics(kafkaProducerConfig.getName(), metricRegistry);
+                                return new KafkaProducerWrapper(kafkaProducerConfig.getDefaultTopic(), kafkaProducerConfig.getName(), kafkaProducer, kafkaProducerMetrics);
+                            }));
+
+            List<SpanKeyExtractor> spanKeyExtractors = SpanKeyExtractorLoader.getInstance().getSpanKeyExtractor(projectConfiguration.getSpanExtractorConfigs());
+            SpanKeyExtractor defaultExtractor = new JsonExtractor();// default extractor for backward compatibility
+            defaultExtractor.configure(projectConfiguration.getSpanExtractorConfigs().get(defaultExtractor.name()));
+            spanKeyExtractors.add(defaultExtractor);// adding default extractor for backward compatibility
```

Comment:
In certain cases, we might not at all need the default extractor. Instead, just an overridden one

Reply:
agreed, this change helps for backward compatibility. Also when we need only overridden one, we can control that in the config file by leaving the producers list as empty